src/main/org/codehaus/groovy/classgen/AsmClassGenerator.java

```diff
@@ -1218,7 +1218,7 @@ public void visitVariableExpression(VariableExpression expression) {
 
         BytecodeVariable variable = controller.getCompileStack().getVariable(variableName, false);
         if (variable == null) {
-            processClassVariable(variableName);
+            processClassVariable(expression);
```

Comment:
I am feeling a bit partial about the change from variableName to expression. I know you need it to set the source position later on. But should we start doing some funky stuff where variableName or expression are changed, the this will cause a problem we may not even notice correctly. So I would suggest something I really really do not often... make the two final